Index: tools/grit/resource_ids
===================================================================
--- tools/grit/resource_ids	(revision 91424)
+++ tools/grit/resource_ids	(working copy)
@@ -41,6 +41,9 @@
   "webkit/tools/test_shell/test_shell_resources.grd": {
     "includes": [5000],
   },
+  "cef/cef_resources.grd": {
+    "includes": [5000],
+  },
   "ui/resources/ui_resources.grd": {
     "includes": [5500],
   },
