Index: src/client/windows/handler/exception_handler.h
===================================================================
--- src/client/windows/handler/exception_handler.h	(revision 699)
+++ src/client/windows/handler/exception_handler.h	(working copy)
@@ -310,7 +310,10 @@
 
   HMODULE dbghelp_module_;
   MiniDumpWriteDump_type minidump_write_dump_;
+
+ public:
   MINIDUMP_TYPE dump_type_;
+ private:
 
   HMODULE rpcrt4_module_;
   UuidCreate_type uuid_create_;