Index: net/proxy/proxy_service.cc
===================================================================
--- net/proxy/proxy_service.cc	(revision 91424)
+++ net/proxy/proxy_service.cc	(working copy)
@@ -452,7 +452,7 @@
   DCHECK(proxy_config_service);
 
   if (!ProxyResolverFactoryForSystem::IsSupported()) {
-    LOG(WARNING) << "PAC support disabled because there is no "
+    DLOG(WARNING) << "PAC support disabled because there is no "
                     "system implementation";
     return CreateWithoutProxyResolver(proxy_config_service, net_log);
   }
Index: cef/libcef/browser_webkit_init.h
===================================================================
--- cef/libcef/browser_webkit_init.h	(revision 1048)
+++ cef/libcef/browser_webkit_init.h	(working copy)
@@ -67,11 +67,17 @@
     WebKit::WebRuntimeFeatures::enableDeviceMotion(false);
     WebKit::WebRuntimeFeatures::enableDeviceOrientation(false);
 
+#ifdef OS_LINUX
+    // Load only the libraries we want Desura to use
+    media::InitializeMediaLibrary("./lib/");
+    WebKit::WebRuntimeFeatures::enableMediaPlayer(true);
+#else
     // Load libraries for media and enable the media player.
     FilePath module_path;
     WebKit::WebRuntimeFeatures::enableMediaPlayer(
-      PathService::Get(base::DIR_MODULE, &module_path) &&
-      media::InitializeMediaLibrary(module_path));
+        PathService::Get(base::DIR_MODULE, &module_path) &&
+        media::InitializeMediaLibrary(module_path));
+#endif
 
     // Construct and initialize an appcache system for this scope.
     // A new empty temp directory is created to house any cached
Index: net/proxy/proxy_service.cc
===================================================================
--- net/proxy/proxy_service.cc	(revision 91424)
+++ net/proxy/proxy_service.cc	(working copy)
@@ -452,7 +452,7 @@
   DCHECK(proxy_config_service);
 
   if (!ProxyResolverFactoryForSystem::IsSupported()) {
-    LOG(WARNING) << "PAC support disabled because there is no "
+    DLOG(WARNING) << "PAC support disabled because there is no "
                     "system implementation";
     return CreateWithoutProxyResolver(proxy_config_service, net_log);
   }
Index: net/ocsp/nss_ocsp.cc
===================================================================
--- net/ocsp/nss_ocsp.cc	(revision 91424)
+++ net/ocsp/nss_ocsp.cc	(working copy)
@@ -578,7 +578,7 @@
   net::URLRequestContext* request_context = g_request_context;
   pthread_mutex_unlock(&g_request_context_lock);
   if (request_context == NULL) {
-    LOG(ERROR) << "No URLRequestContext for OCSP handler.";
+    DLOG(ERROR) << "No URLRequestContext for OCSP handler.";
     // The application failed to call SetURLRequestContextForOCSP, so we
     // can't create and use net::URLRequest.  PR_NOT_IMPLEMENTED_ERROR is not an
     // accurate error code for this error condition, but is close enough.

