diff --git algparam.h algparam.h
index ea5129c..ec2a175 100644
--- algparam.h
+++ algparam.h
@@ -319,7 +319,7 @@ public:
 
 	void MoveInto(void *buffer) const
 	{
-		AlgorithmParametersTemplate<T>* p = new(buffer) AlgorithmParametersTemplate<T>(*this);
+		new(buffer) AlgorithmParametersTemplate<T>(*this);
 	}
 
 protected:
diff --git misc.h misc.h
index 8425c53..3bd99b5 100644
--- misc.h
+++ misc.h
@@ -55,7 +55,7 @@ struct CompileAssert
 #if defined(CRYPTOPP_EXPORTS) || defined(CRYPTOPP_IMPORTS)
 #define CRYPTOPP_COMPILE_ASSERT_INSTANCE(assertion, instance)
 #else
-#define CRYPTOPP_COMPILE_ASSERT_INSTANCE(assertion, instance) static CompileAssert<(assertion)> CRYPTOPP_ASSERT_JOIN(cryptopp_assert_, instance)
+#define CRYPTOPP_COMPILE_ASSERT_INSTANCE(assertion, instance)
 #endif
 #define CRYPTOPP_ASSERT_JOIN(X, Y) CRYPTOPP_DO_ASSERT_JOIN(X, Y)
 #define CRYPTOPP_DO_ASSERT_JOIN(X, Y) X##Y
diff --git secblock.h secblock.h
index 24b9fc0..6fed9de 100644
--- secblock.h
+++ secblock.h
@@ -29,7 +29,7 @@ public:
 
 	pointer address(reference r) const {return (&r);}
 	const_pointer address(const_reference r) const {return (&r); }
-	void construct(pointer p, const T& val) {new (p) T(val);}
+	void construct(pointer p, const T& val = T()) {new (p) T(val);}
 	void destroy(pointer p) {p->~T();}
 	size_type max_size() const {return ~size_type(0)/sizeof(T);}	// switch to std::numeric_limits<T>::max later
 
