Index: SConstruct
===================================================================
--- SConstruct	(revision 8848)
+++ SConstruct	(working copy)
@@ -242,6 +242,7 @@
     },
     'mode:debug': {
       'CCFLAGS':      ['/Od', '/Gm'],
+	  'CXXFLAGS':     ['/Fd"v8-d.pdb"'],
       'CPPDEFINES':   ['_DEBUG', 'ENABLE_DISASSEMBLER', 'DEBUG'],
       'LINKFLAGS':    ['/DEBUG'],
       'msvcrt:static': {
@@ -253,6 +254,7 @@
     },
     'mode:release': {
       'CCFLAGS':      ['/O2'],
+	  'CXXFLAGS':     ['/Fd"v8.pdb"'],
       'LINKFLAGS':    ['/OPT:REF', '/OPT:ICF'],
       'msvcrt:static': {
         'CCFLAGS': ['/MT']
@@ -312,7 +314,8 @@
   },
   'msvc': {
     'all': {
-      'WARNINGFLAGS': ['/W3', '/WX', '/wd4351', '/wd4355', '/wd4800']
+	  #'/WX', removed by mark
+      'WARNINGFLAGS': ['/W3', '/wd4351', '/wd4355', '/wd4800']  
     },
     'library:shared': {
       'CPPDEFINES': ['BUILDING_V8_SHARED'],
@@ -829,7 +832,8 @@
 
 SUFFIXES = {
   'release': '',
-  'debug': '_g'
+  'debug': '-d',
+  'static': '-s'
 }
 
 
@@ -1328,6 +1332,10 @@
   # Generate library base name.
   target_id = mode
   suffix = SUFFIXES[target_id]
+  
+  if env['msvcrt'] == 'static':
+	suffix = SUFFIXES['static']
+  
   library_name = 'v8' + suffix
   preparser_library_name = 'v8preparser' + suffix
   version = GetVersion()
